ibis/mapd/operations.py

```diff
@@ -54,8 +54,20 @@ def _cast(translator, expr):
     op = expr.op()
     arg, target = op.args
     arg_ = translator.translate(arg)
-    type_ = str(MapDDataType.from_ibis(target, nullable=False))
 
+    if isinstance(arg, ir.GeoSpatialValue):
+        # NOTE: CastToGeography expects geometry with SRID=4326
+        dtype_from = arg._dtype.geotype
+        dtype_target = target.geotype
+
+        if dtype_from == dtype_target:
```

Comment:
Why is this an error and not a no-op?

Reply:
I think I am not following you here. could you provide more details?

Reply:
I think I'm not understanding why it's an error to cast from a geotype to the same geotype. That is what you're doing here is it not?

Reply:
Oh I see .. you're right. I will fix that! thanks!